src/fastalloc/iter.rs

```diff
@@ -0,0 +1,643 @@
+use crate::{Operand, OperandKind, OperandPos, OperandConstraint};
+
+#[derive(Clone, Copy, PartialEq)]
+enum OperandConstraintKind {
+    Any,
+    Reg,
+    Stack,
+    FixedReg,
+    Reuse,
+}
+
+impl PartialEq<OperandConstraint> for OperandConstraintKind {
```

Comment:
Rather than this impl, can we implement `From<OperandConstraint> for OperandConstraintKind`, then one could do `constraint.kind() == OperandConstraintKind::Reg`? This is slightly more general and cleaner in my experience...